nnpdf_data/nnpdf_data/theorydbutils.py

```diff
@@ -51,13 +49,35 @@ def fetch_theory(theory_database: Path, theoryID: int):
     >>> from nnpdf_data.theorydbutils import fetch_theory
     >>> theory = fetch_theory(theory_cards, 700)
     """
-    filepath = theory_database / f"{theoryID}.yaml"
-    tdict = parse_theory_card(filepath)
+
+    available_theories = get_available_theory_cards(theory_database)
+    try:
+        theoryfile = available_theories[theoryID]
+    except KeyError as e:
+        raise TheoryNotFoundInDatabase(f"Theorycard for theory not found: {e}")
+
+    tdict = parse_theory_card(theoryfile)
     if tdict["ID"] != int(theoryID):
-        raise ValueError(f"The theory ID in {filepath} doesn't correspond with its ID entry")
+        raise ValueError(f"The theory ID in {theoryfile} doesn't correspond with its ID entry")
     return tdict
 
 
+@lru_cache
+def get_available_theory_cards(path):
+    """Since theoryfile names may contain underscores, the theoryfile name cannot uniquely be
+    determined from the theoryID. Therefore we create a mapping between the theoryid as python
+    int and the corresponding file.
+    """
+    available_theories = {}
+    for theoryfile in path.glob("*.yaml"):
+        tmp_id = int(theoryfile.name.removesuffix('.yaml'))
```

Comment:
```suggestion
        tmp_id = int(theoryfile.stem)
```

would this work? (nitpicking here)